### PR TITLE
kubeadm: fix the generation of ECDSA keys in kubeconfig files

### LINKED ISSUE #3072: kubeadm init phase kubeconfig does not respect PublicKeysECDSA
## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`):
```
kubeadm version: &version.Info{Major:"1", Minor:"30", GitVersion:"v1.30.0", GitCommit:"7c48c2bd72b9bf5c44d21d7338cc7bea77d0ad2a", GitTreeState:"archive", BuildDate:"2024-06-05T13:36:39Z", GoVersion:"go1.22.4", Compiler:"gc", Platform:"linux/arm64"}
```

## What happened?
The ```client-key-data``` is a RSA key even ```PublicKeysECDSA``` feature gate is set to true.

## What you expected to happen?
The key should be a ECDSA key

## How to reproduce it (as minimally and precisely as possible)?
```
[root@lima-alpine ~]# cat << EOF > /tmp/cluster.yaml
apiVersion: kubeadm.k8s.io/v1beta3
kind: ClusterConfiguration
featureGates:
  PublicKeysECDSA: true
apiServer:
  certSANs: ['localhost', '127.0.0.1']
EOF
[root@lima-alpine ~]# kubeadm init --config /tmp/cluster.yaml phase kubeconfig admin
[kubeconfig] Writing "admin.conf" kubeconfig file
```

## Anything else we need to know?
N/A

### PR DESCRIPTION
**Pull request by neolit123**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

When the PublicKeysECDSA feature gate is used or the new v1beta4.ClusterConfiguration.EncryptionAlgorithm field is used with "ECDSA-P256" as value, make sure that this is reflected in the "cert spec" used to generate private keys and they end up as "EC keys".

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubeadm/issues/3072

#### Does this PR introduce a user-facing change?

```release-note
kubeadm: fixed a bug where the PublicKeysECDSA feature gate was not respected when generating kubeconfig files.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/phases/kubeconfig/kubeconfig_test.go [hunk 1]
```diff
@@ -180,6 +181,11 @@ func TestGetKubeConfigSpecs(t *testing.T) {
 					t.Errorf("getKubeConfigSpecs for %s Organizations is %v, expected %v", assertion.kubeConfigFile, spec.ClientCertAuth.Organizations, assertion.organizations)
 				}
 
+				// Assert EncryptionAlgorithm
+				if spec.EncryptionAlgorithm != cfg.EncryptionAlgorithm {
```
Comment (carlory) [inline cmd/kubeadm/app/phases/kubeconfig/kubeconfig_test.go:185]: Is the feature-gate `PublicKeysECDSA` still available in v1.31? If yes, adding a new test case to check the result when the feature gate is enabled is better. And the expected EncryptionAlgorithm should be obtained via `cfg.ClusterConfiguration.EncryptionAlgorithmType()` until the feature-gate is removed.

Comment (neolit123) [reply cmd/kubeadm/app/phases/kubeconfig/kubeconfig_test.go:185]: @carlory i have added a second commit here that adds the missing UT

**Comment by pacoxu** [comment]
/hold 
@carlory  can you take a look again?

**Pull request merged**